Makefile

```
@@ -372,23 +373,50 @@ install-helm:
 	tar -xvzf /tmp/helm.tar.gz --directory=${GOPATH}/bin --strip-components=1 $(OS)-$(OS_ARCH)/helm
 	rm /tmp/helm.tar.gz
 
+# delete controller-gen
+delete-controller-gen:
+ifeq (,$(shell which controller-gen))
+	$(info controller-gen is not installed)
+else
+	rm $(shell which controller-gen)
+endif
+
 # find or download controller-gen
 # download controller-gen if necessary
 install-controller-gen:
 ifeq (,$(shell which controller-gen))
+	$(info installing controller-gen...)
 	@{ \
 	set -e ;\
 	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$CONTROLLER_GEN_TMP_DIR ;\
 	go mod init tmp ;\
-	CGO_ENABLED=0 go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.3 ;\
+	CGO_ENABLED=0 go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION) ;\
 	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
 	}
 CONTROLLER_GEN=$(GOBIN)/controller-gen
 else
+ifneq ($(shell controller-gen --version | sed "s/Version:\ //g"), $(CONTROLLER_GEN_VERSION))
+	$(info Update your controller-gen with update-controller-gen target)
+	$(info Your Version:    $(shell controller-gen --version | sed "s/Version://g"))
+	$(info Required Version: $(CONTROLLER_GEN_VERSION))
+	exit -1
```

An alternative would be to silently update to desired version by calling MAKE inside make::
```suggestion
ifneq ($(shell controller-gen --version | sed "s/Version:\ //g"), $(CONTROLLER_GEN_VERSION))
	$(info Update your controller-gen with update-controller-gen target)
	$(info Your Version:    $(shell controller-gen --version | sed "s/Version://g"))
	$(info Required Version: $(CONTROLLER_GEN_VERSION))
	$(MAKE) update-controller-gen
```

Seems definitely OK to keep what you as we can suppose using our repositories are using other version, hence it's better for them to be aware we are not at the same version 🤗